src/views/catalog/wizard/components/WizardFooter.tsx

```diff
@@ -59,12 +64,18 @@ export const WizardFooter: React.FC<{ namespace: string }> = ({ namespace }) =>
   const templateNamespace = tabsData?.overview?.templateMetadata?.namespace;
   const loaded = vmContextLoaded && vmCreateLoaded && bootSourceLoaded;
 
-  React.useEffect(() => {
-    if (loaded && startVM !== isBootSourceAvailable) {
+  useEffect(() => {
+    updateTabsData((tabsDataDraft) => {
+      tabsDataDraft.startVM = startVM;
+    });
+  }, [startVM, updateTabsData]);
+
+  useEffect(() => {
+    if (loaded && startVM && !isBootSourceAvailable) {
+      // change startVM to false only if boot source is not available
       setStartVM(isBootSourceAvailable);
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [loaded, isBootSourceAvailable]);
+  }, [loaded, isBootSourceAvailable, startVM]);
```

Comment:
pretty sure this useEffect is not needed anymore and can be removed

Reply:
What is the reason of being so sure about removing `useEffect` in here?

Reply:
I'd leave `useEffect` in there, maybe also would better add `useCallback` to `onChangeStartVM`. WDYT?

Reply:
let's focus on the isBootSourceAvailable and startVM.
isBootSourceAvailable - is set to true if the boot source is available and we can see both the start vm checkbox and the quick create vm.
startVM - is now getting the carried value from the checkbox at the drawer, this checkbox exists only when we can see the quick create button - when the boot source is available, which means if the boot source is not available, this checkbox has a falsy value which will give you the exact same thing as the effect is doing, which causing to un-needed renders.

Reply:
I think I am not getting it. From the code of _src/views/catalog/wizard/components/WizardFooter.tsx_, I can see the checkbox can be displayed no matter `isBootSourceAvailable` in this file. And according to the code above (the original useEffect block or the edited one), `isBootSourceAvailable` can be false. I still think we need to update `startVM` value to the correct one, if that changes to `true` while `isBootSourceAvailable` is false.
The best solution I can see for this is to completely remove this `useEffect` block while setting the initial value of `startVM` according to `isBootSourceAvailable` value, and also updating `isDisabled` param of the `Checkbox` accodfing to `isBootSourceAvailable`, too. WDYT?

Reply:
I wouldn't rely on that condition `canQuickCreate` in [TemplatesCatalogDrawerCreateForm-L.101](https://github.com/kubevirt-ui/kubevirt-plugin/blob/893005c63fd46134da92cb176991b00319673c75/src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx#L101). Because if there will be some changes and the component will be moved, it stops working as expected.

Reply:
we shouldn't rely on things that may happen in the future or may not.
bottom line this useEffect is not needed and costs performance and component complexity